operator/internal/handlers/lokistack_create_or_update.go

```diff
@@ -101,7 +101,7 @@ func CreateOrUpdateLokiStack(
 
 	objStore.Schemas = storageSchemas
 
-	if stack.Spec.Storage.TLS != nil {
+	if stack.Spec.Storage.TLS != nil && stack.Spec.Storage.TLS.CA != "" {
```

Comment:
Should we set [the validation](https://github.com/periklis/loki/blob/82161fe329f6aef99c52368a47a30f096887d14e/operator/apis/loki/v1/lokistack_types.go#L361-L362) on the CRD attribute to "required"?

Should there be a "degraded error" when TLS is set but CA is empty?

Reply:
> Should we set [the validation](https://github.com/periklis/loki/blob/82161fe329f6aef99c52368a47a30f096887d14e/operator/apis/loki/v1/lokistack_types.go#L361-L362) on the CRD attribute to "required"?

I think you are right we should set the field `CA` to required in the first place. Means whenever someone creates the spec block, they will be required to enter a configmap (i.e. in the OpenShift Console uncollapsing the block creates the block in the first place). 

> Should there be a "degraded error" when TLS is set but CA is empty?

The handler does already lookup for the ConfigMap first and if not found (includes the empty name case) sets the resource to a degraded condition.

Reply:
Need to circle back to the original solution because even if `ObjectStorageTLSSpec` is uncollapsed the console renderer makes the `caName` field required, i.e.:
![image](https://user-images.githubusercontent.com/152312/203721877-0f481ca6-8f77-41ef-9c9b-67ff73a0238c.png)
![image](https://user-images.githubusercontent.com/152312/203721935-26b98860-6a89-41ca-9ad4-542ce4e4a7c1.png)
![image](https://user-images.githubusercontent.com/152312/203722009-bfcb7b52-306c-4bc6-bcfe-f7457f80cbca.png)

This is really an unfortunate combination where we need to keep the default on `caKey` but make the entire struct optional. :(

Reply:
Hm, I assumed that the editor would not add the whole `tls` block unless the user actively added it, because that is marked as `optional` as well. An alternative/extension would be to remove the default from the `caKey` and do the default-handling in code instead.

Reply:
There is more to add here, i.e. the handling of degraded errors is broken too :(. But making the `caKey` optional and move the default back to code is a good option.